src/query/models/tag.go

```diff
@@ -145,17 +146,46 @@ func (m Matchers) ToTags() (Tags, error) {
 	return Normalize(tags), nil
 }
 
-// ID returns a string representation of the tags
-func (t Tags) ID() string {
-	b := make([]byte, 0, len(t))
+// StringID returns a string representation of the tags
+func (t Tags) StringID() string {
```

Comment:
(unrelated to the pr)

Tags keys/values are allowed to have '=' and ','; this encoding isn't invertible - i.e. given a string representation of some tags, you can't guarantee that it comes from a single value.  

How is this function used?

Reply:
This determines what the ID of the metric will be in M3DB

Reply:
we should talk about this offline.